packages/components/src/navigation/index.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { Icon, arrowLeft } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
+import Text from '../text';
+import Item from './item';
+
+const Navigation = ( { data, initial } ) => {
+	const initialActive = data.find( ( item ) => item.slug === initial );
+	const [ active, setActive ] = useState( initialActive );
+	const parent = data.find( ( item ) => item.slug === active.parent );
+	const items = data.filter( ( item ) => item.parent === active.parent );
+
+	const goBack = () => {
+		if ( ! parent.parent ) {
+			// We are at top level, will need to handle this case.
+			return;
+		}
+		const parentalSiblings = data.filter(
+			( item ) => item.parent === parent.parent
+		);
+		if ( parentalSiblings.length ) {
+			setActive( parentalSiblings[ 0 ] );
+		}
+	};
+
+	return (
+		<div className="components-navigation">
+			<Button className="components-navigation-back" onClick={ goBack }>
+				<Icon icon={ arrowLeft } />
+				{ parent.back }
+			</Button>
+			<div className="components-navigation-title">
+				<Text variant="title.medium">{ parent.title }</Text>
+			</div>
+			<div className="components-navigation-items">
+				{ items.map( ( item ) =>
+					! item.isSecondary ? (
+						<Item
```

Comment:
Good question. From the prototype, it appears the primary v. secondary dichotomy only exists at the top level. Can you confirm this @jameskoster ? In other words, once you being drilling down, it will a become a single menu and there won't be a secondary section.

I added a multiple depth level to illustrate multiple levels of depth:

![depth](https://user-images.githubusercontent.com/1922453/88245978-277c6280-cced-11ea-8344-e6058d82de17.gif)

Reply:
> In other words, once you being drilling down, it will a become a single menu and there won't be a secondary section.

That was the case for WooCommerce. I cannot say that it will also be the case for all other plugins, or core for that matter. Looking at the [latest designs](https://github.com/WordPress/gutenberg/issues/23939) for core there seem to be separate sections within a drilldown (note 'recent' and 'drafts' sections):

![sections](https://cldup.com/oXG3SfY9v7.png)

I'm not sure if they qualify as primary / secondary though... And these designs are still being worked on, so this should probably be taken with a grain of salt :D

Reply:
>I cannot say that it will also be the case for all other plugins

Ok, thanks for the clarification. I think that means every menu item will need a distinction, ie `menu: 'primary|secondary'` as suggested by @joshuatf. Probably defaulting to `primary`.

That is of course unless we want to allow unlimited groupings 🤔 . For now, assuming a maximum of two sections will work and we can revamp later.

Reply:
If we go forward with @ItsJonQ's suggestion then I think we will end up removing `primary` and `secondary` logic altogether and this can be determined by the navigation plugin consuming this component.